flutter_assignment/lib/features/home_page/data/datasource/movies_remote_datasource.dart

```diff
@@ -1,42 +1,41 @@
 import 'dart:convert';
 
+import 'package:dartz/dartz.dart';
+import 'package:flutter_assignment/constants/constants.dart';
 import 'package:flutter_assignment/core/error/exceptions.dart';
 import 'package:flutter_assignment/features/home_page/data/models/star_war_model.dart';
 import 'package:http/http.dart' as http;
 
-abstract class MoviesRemoteDatasource {
-  /// Calls the Local endpoint
-  ///
-  /// Throws [ServerExceptions] exception if error occurs
-  // Future<StarWarMoviesModel> collectMoviesListFromLocal();
+import '../../../../core/error/failure.dart';
 
+abstract class MoviesRemoteDatasource {
   /// Calls the API endpoint
   ///
   /// Throws [ServerExceptions] exception if error occurs
-  Future<StarWarMoviesModel> collectMoviesListFromAPI(String url);
+  Future<Either<Failure, StarWarMoviesModel>> collectMoviesListFromAPI(String url);
 }
 
 class RemoteDataSourceImpl implements MoviesRemoteDatasource {
   final http.Client client;
 
   RemoteDataSourceImpl({required this.client});
 
+  /// This Function helps to collect movies data from the swapi.dev/films API
   @override
-  Future<StarWarMoviesModel> collectMoviesListFromAPI(String url) async {
```

Comment:
Please add proper comments and user proper function name for better understanding.